### PR TITLE
FEI-4966.8: Update eslint configuration and fix lint

### PR DESCRIPTION
**Pull request by kevinbarabash**
## Summary:
Some of the lint fixes were done using `yarn lint --fix` while other where done using find'n'replace to update rule names in `eslint-disable` comments.  The rest were manual changes.  The next PRs will update the babel and jest configs so that we can ensure that things build and that tests are passing.

TODO:
- [x] tweak some of the disabled rules to be disable for test files only

Issue: FEI-4966

## Test plan:
- yarn lint
- yarn typecheck

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -88,7 +97,7 @@ module.exports = {
         "react/react-in-jsx-scope": "error",
         "react/require-render-return": "error",
         "monorepo/no-internal-import": "error",
-        "monorepo/no-relative-import": "error",
+        // "monorepo/no-relative-import": "error",
```
Comment (jeresig) [inline .eslintrc.js:100]: Might want to copy the comment for this from wonder-stuff.

File: packages/wonder-blocks-banner/src/components/__docs__/banner.stories.tsx [hunk 2]
```diff
@@ -12,6 +11,7 @@ import {LabelSmall} from "@khanacademy/wonder-blocks-typography";
 
 // @ts-expect-error [FEI-5019] - TS2305 - Module '"@storybook/react"' has no exported member 'StoryComponentType'.
 import type {StoryComponentType} from "@storybook/react";
+import Banner from "@khanacademy/wonder-blocks-banner";
```
Comment (jeresig) [inline packages/wonder-blocks-banner/src/components/__docs__/banner.stories.tsx:14]: Why the move?

**Review by jeresig — APPROVED** [review]
Lots of fixes! Thanks for resolving all of this.

File: packages/wonder-blocks-banner/src/components/__docs__/banner.stories.tsx (hunk 2 above)
Comment (kevinbarabash) [reply packages/wonder-blocks-banner/src/components/__docs__/banner.stories.tsx:14]: Bad autofixing from `import/order`.  I can fix this.

File: .eslintrc.js [hunk 3]
```diff
@@ -30,17 +39,32 @@ module.exports = {
...
+        // from node_modules/@types/react/index.d.ts
+        JSX: false,
+        // from node_modules/typescript/lib/lib.dom.d.ts
+        DOMHighResTimeStamp: false,
+        RequestInfo: false,
+        RequestInit: false,
+        // from types/utility.d.ts
+        Empty: false,
+        ObjMap: false,
+        SpreadType: false,
```
Comment (kevinbarabash) [inline .eslintrc.js:62]: Some people suggest disable `no-undef` all the time, e.g. https://stackoverflow.com/questions/65054079/eslint-with-typescript-and-firebase-no-undef-error.  That doesn't seem like a good solution.  Eslint should be able to tell the difference between value identifier and type identifiers and ignore just the undefined type identifiers.

**Pull request merged**